lib/three_scale/domain_substitution.rb

```diff
@@ -131,6 +125,13 @@ def external_self_domain
       ThreeScale::DomainSubstitution::Substitutor.to_external(self['self_domain'])
     end
 
+    # This is just an alias to _#self_domain_
+    # @deprecated Use {#internal_admin_domain}
+    # @return [String] the database _self_domain_ value
+    def internal_self_domain
+      self['self_domain']
+    end
```

Comment:
Do we need this method still? Can it be removed?

Reply:
This method is supposed to be changed to `internal_admin_domain`, which calls `admin_domain`. To answer the question we need to take a look at its definition:
https://github.com/3scale/porta/blob/898e411daeb8f7e9eeea4d9df5bdd3a3f64557a7/app/models/account/provider_methods.rb#L327-L337

It's basically the same problem we have with `#external_self_domain`. This change will throw errors when called by a non provider. Can't remember where this happens but we'll need to fix that before deleting this 2 methods.

Reply:
I think that one can just call `self_domain` if this is needed. Not externally though. `self_domain` should be private if we don't want it to be used externally.